openhands/runtime/action_execution_server.py

```diff
@@ -76,6 +76,39 @@ class ActionRequest(BaseModel):
 api_key_header = APIKeyHeader(name='X-Session-API-Key', auto_error=False)
 
 
+def is_binary_file(file_path):
+    """
+    Check if a file is binary.
+
+    Args:
+        file_path (str): Path to the file to check.
+
+    Returns:
+        bool: True if the file is binary, False otherwise.
+    """
+    # Number of bytes to check at the beginning of the file
+    sample_size = 8192
+
+    try:
+        # Open the file in binary mode
+        with open(file_path, 'rb') as f:
+            sample = f.read(sample_size)
+
+        # Check for NULL bytes (common in binary files)
+        if b'\x00' in sample:
+            return True
+
+        try:
+            sample.decode('utf-8')
+            return False
+        except UnicodeDecodeError:
+            # If decoding fails, it's likely binary
+            return True
+
+    except IOError:
+        return None
```

Comment:
This probably handles enough cases for our purposes, but if we're going to implement it here we should have tests around it.

What might be easier, I notice we have the `binaryornot` package in our dependency lock file, so maybe this is a good opportunity to use that and upgrade it to a direct dependency in `pyproject.toml`. 

https://github.com/binaryornot/binaryornot/tree/main